app/assets/javascripts/admin/addon/components/color-palette-editor.gjs

```diff
@@ -113,22 +119,42 @@ const Picker = class extends Component {
     }
   }
 
+  @action
+  onTextPaste(event) {
+    const content = (event.clipboardData || window.clipboardData).getData(
+      "text"
+    );
+
+    if (!this.isValidHex(content)) {
+      event.preventDefault();
+      this.toasts.error({
+        data: {
+          message: i18n(
+            "admin.config_areas.color_palettes.invalid_color_length"
+          ),
+        },
+      });
+      return;
+    }
+  }
```

Comment:
Having to handle all these events manually feels odd to me. In my mind, the pattern should be that each input tracks its own "value" and reacts accordingly when the value of its input changes.

Reply:
I think this is related to the toasts comment above. I'm using toasts because we don't have an outlet/space in the palette editor to show errors for individual colors. If we change our design to make space for errors, I think we'll be able to use the "tracking" pattern you describe here.

Reply:
I see. Let's defer this but we should definitely follow up with product/design about this.